### PR TITLE
Moved kernel userland code into the osquery directory structure.

### PR DESCRIPTION
**Pull request by mofarrell**
The makefile will no longer effectively build the test system.
The kernel/user directory is on its way out.

### CODE REVIEW

File: osquery/events/kernel/circular_queue_user.cpp [hunk 1]
```diff
@@ -100,3 +102,5 @@ int CQueue::kernelSync() {
   return update.drops;
 }
 
+}  // namespace osquery
+
```
Comment (theopolis) [inline osquery/events/kernel/circular_queue_user.cpp]: nit, ditch this last blank line after the namespace close, just end with a newline.

File: osquery/events/tests/producer.cpp [hunk 2]
```diff
@@ -8,7 +8,10 @@
  *
```
Comment (theopolis) [inline osquery/events/tests/producer.cpp]: I would recommend naming these producer/consumer tests something related to our kernel buffering/eventing implementation: `osquery/events/tests/kernel_{producer,consumer}.cpp`.

Comment (mofarrell) [inline osquery/events/tests/producer.cpp]: What about just moving them to `osquery/events/kernel/tests/`  kind of the way the other test directories are like `osquery/events/linux/tests`

Also worth noting they are going to disappear soon.

Comment (theopolis) [inline osquery/events/tests/producer.cpp]: Sounds good!

File: osquery/events/tests/consumer.cpp [hunk 3]
```diff
@@ -8,7 +8,10 @@
  *
  */
 
-#include <circular_queue_user.h>
+#undef DEBUG
+#ifdef DEBUG
```
Comment (theopolis) [inline osquery/events/tests/consumer.cpp]: Undef then check if def?

**Pull request merged**